crates/cubecl-linalg/src/convolution/loader/im2col_tma.rs

```diff
@@ -80,30 +81,78 @@ impl<MP: MatmulPrecision, G: ConvGemmConfig> TmaIm2colLoader<MP, G> {
             let mut full_stage = stage.as_slice_mut(1u32);
             let tensor = this.map.tensor.try_cast_unchecked();
 
-            let in_h = (this.map.h_offset * config.stride(0)) as i32 - config.padding(0);
-            let in_w = (this.map.w_offset * config.stride(1)) as i32 - config.padding(1);
+            let spatial_dims = comptime![this.map.spatial_offsets.len()];
+            let mut in_offs = Sequence::<i32>::new();
+
+            #[unroll]
+            for dim in 0..spatial_dims {
+                let dim = unwrap(dim);
+                let offs = this.map.spatial_offsets.index(dim) * comptime![config.stride(dim)];
+                let offs = offs as i32 - comptime![config.padding(dim)];
+                in_offs.push(offs);
+            }
 
             #[unroll]
             for tile_k in 0..tiling_dims.tile_count_col() {
                 let k = this.map.k_offset + tile_k * k_size;
                 let (k_idx, channel_start) = this.padded_channels.div_mod(k);
-                let (k_x, k_y) = (k_idx % config.kernel_size(1), k_idx / config.kernel_size(1));
                 let slice_start = tile_k * slice_size;
                 let mut stage = full_stage.slice_mut(slice_start, slice_start + slice_size);
 
-                let offset_y = k_y * config.dilation(0);
-                let offset_x = k_x * config.dilation(1);
-
-                bar.tma_load_im2col_4d(
-                    &tensor,
-                    &mut stage,
-                    this.map.n_offset as i32,
-                    in_h,
-                    in_w,
-                    channel_start as i32,
-                    offset_y as u16,
-                    offset_x as u16,
-                );
+                match config.dimensionality() {
+                    Dimensionality::Dim1 => {
+                        let offset = k_idx * config.dilation(0);
+
+                        bar.tma_load_im2col_3d(
```

Comment:
3d means rank 3 (with batch size and channel I guess?) Maybe adding a comment would be great.